pkg/kube/multicluster/secretcontroller.go

```diff
@@ -91,15 +92,36 @@ type Controller struct {
 
 // NewController returns a new secret controller
 func NewController(kubeclientset kube.Client, namespace string, localClusterID cluster.ID) *Controller {
+	// When these two are set to true, Istiod will be watching the namespace in which
+	// Istiod is running on the external cluster. Use the inCluster credentials to
+	// create a kubeclientset
+	informerClient := kubeclientset
+	if features.LocalClusterSecretWatcher && features.ExternalIstiod {
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			log.Errorf("Could not get istiod incluster configuration: %v", err)
+			return nil
+		}
+		log.Info("Successfully retrieved incluster config.")
+
+		localKubeClient, err := kube.NewClient(kube.NewClientConfigForRestConfig(config))
```

Comment:
nit: maybe only init Kubernetes client with k8s.io.client-go

Reply:
+1 - if we move this code close to the other k8s init, also use the same pattern/library/deps.